app-android/src/main/java/com/ustadmobile/port/android/view/composable/CourseBlockEdit.kt

```diff
@@ -0,0 +1,252 @@
+            )
+
+            Text(text = stringResource(id = R.string.penalty_label))
+        }
+    }
+
+}
+
+@Composable
+private fun FirstRow(
+    uiState: CourseBlockEditUiState,
+    onCourseBlockChange: (CourseBlock?) -> Unit
+){
+    Row {
+        Column(
+            modifier = Modifier.weight(0.5F)
+        ) {
+
+            UstadDateEditTextField(
+                value = uiState.courseBlock?.cbHideUntilDate ?: 0,
+                label = stringResource(id = R.string.dont_show_before)
+                    .addOptionalSuffix(),
+                error = uiState.caStartDateError,
+                enabled = uiState.fieldsEnabled,
+                onValueChange = {
+                    onCourseBlockChange(uiState.courseBlock?.shallowCopy{
+                        cbHideUntilDate = it
+                    })
+                }
+            )
+
+            Text("timeZone")
+        }
+
+        Spacer(modifier = Modifier.width(15.dp))
+
+        UstadDateEditTextField(
+            modifier = Modifier.weight(0.5F),
+            value = uiState.courseBlock?.cbHideUntilDate ?: 0,
+            label = stringResource(id = R.string.time),
+            enabled = uiState.fieldsEnabled,
+            onValueChange = {
+                onCourseBlockChange(uiState.courseBlock?.shallowCopy{
+                    cbHideUntilDate = it
+                })
+            }
+        )
+    }
+}
+
+@Composable
+private fun SecondRow(
+    uiState: CourseBlockEditUiState,
+    onCourseBlockChange: (CourseBlock?) -> Unit
+){
+    Row() {
+        UstadMessageIdOptionExposedDropDownMenuField(
+            modifier = Modifier.weight(0.5F),
+            value = uiState.courseBlock?.cbCompletionCriteria ?: 0,
+            label = stringResource(R.string.completion_criteria),
+            options = CompletionCriteriaConstants.COMPLETION_CRITERIA_MESSAGE_IDS,
+            onOptionSelected = {
+                onCourseBlockChange(uiState.courseBlock?.shallowCopy{
+                    cbCompletionCriteria = it.value
+                })
+            },
+            enabled = uiState.fieldsEnabled,
+        )
+
+        Spacer(modifier = Modifier.width(15.dp))
+
+        if (uiState.minScoreVisible){
+            UstadTextEditField(
+                modifier = Modifier.weight(0.5F),
+                value = ((uiState.courseBlock?.cbMaxPoints ?: 0).toString() + " " +
+                        stringResource(id = R.string.points)),
+                label = stringResource(id = R.string.maximum_points),
+                enabled = uiState.fieldsEnabled,
+                onValueChange = {
+                    onCourseBlockChange(uiState.courseBlock?.shallowCopy {
+                        cbMaxPoints = it.toInt()
+                    })
+                }
+            )
+        }
+    }
+}
+
+@Composable
+private fun FourthRow(
+    uiState: CourseBlockEditUiState,
+    onCourseBlockChange: (CourseBlock?) -> Unit
+){
+    Row {
+        UstadDateEditTextField(
```

Comment:
There will be one new component UstadDateTimeEditTextFields - you can switch to using that. Then one composable will include both date and time.